[gordonajajar] iP

gordonajajar opened the pull request.

gordonajajar committed:
Add increment: level-0

gordonajajar committed:
Add increment: level-1

gordonajajar committed:
Add increment: level-2

gordonajajar committed:
Add increment: level-3

LJQ2001 commented on src/main/java/Bruh.java:
```diff
@@ -11,16 +12,29 @@ public static void main(String[] args) {
+            line = in.nextLine();
+
             System.out.println("-----------------------------------------------------------------------------");
             if (line.equals("bye")) {
```
Great use in Egyptian Style format for conditional Brackets !!!

LJQ2001 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,35 @@
+public class Task {
```
Add a header comment for the Task Class

LJQ2001 commented on src/main/java/Bruh.java:
```diff
@@ -26,12 +26,21 @@ public static void main(String[] args) {
                 int idx = 0;
-                for (int i = 0; i < listIndex; i++) {
-                    System.out.println(i + 1 + ". " + list[i]);
+                for (int i = 0; i < taskIndex; i++) {
```
Maybe add comments for the conditional statement (Brief Description on what case does what)

LJQ2001 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
```
This comment is probably redundant as its stating the obvious.

abefqy replied:
Mostly redundant comment

sean6369 replied:
Unnecessary comment

LJQ2001 reviewed (commented):
Good Job

abefqy commented on src/main/java/Bruh.java:
```diff
@@ -0,0 +1,51 @@
+                tasks[markIndex].setDone(true);
+                System.out.println("Marked as done: [X] " + tasks[markIndex].getDescription());
+            } else if (line.startsWith("unmark ")) {
+                int unmarkIndex = -1 + Integer.parseInt(line.substring(7));
```
```suggestion
                int unmarkIndex = Integer.parseInt(line.substring(7)) -1;
```

abefqy replied:
Perhaps shift the -1 to the end of the line for greater readability

abefqy commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,35 @@
+
+
+    protected String description;
+    protected boolean isDone;
```
Any specific reason for using protected instead of private?

abefqy commented on src/main/java/Bruh.java:
```diff
@@ -0,0 +1,51 @@
+            } else if (line.startsWith("mark ")) {
+                int markIndex = -1 + Integer.parseInt(line.substring(5));
+                tasks[markIndex].setDone(true);
+                System.out.println("Marked as done: [X] " + tasks[markIndex].getDescription());
```
Can consider using 2 different methods for markDone() and markUndone()

abefqy reviewed (commented):
Good work! Minute changes suggested.

sean6369 commented on src/main/java/Bruh.java:
```diff
@@ -0,0 +1,51 @@
+                System.out.println("Here are your tasks: ");
+                int idx = 0;
+                for (int i = 0; i < taskIndex; i++) {
+                    System.out.println(i + 1 + ". [" + tasks[i].getStatusIcon() + "]" + tasks[i].getDescription());
```
can create a toString for the task class such that printing tasks is simpler

sean6369 commented on src/main/java/Bruh.java:
```diff
@@ -0,0 +1,51 @@
+                break;
+            } else if (line.equals("list")) {
+                System.out.println("Here are your tasks: ");
+                int idx = 0;
```
what is the use of idx?

sean6369 commented on src/main/java/Bruh.java:
```diff
@@ -0,0 +1,51 @@
+                    System.out.println(i + 1 + ". [" + tasks[i].getStatusIcon() + "]" + tasks[i].getDescription());
+                }
+            } else if (line.startsWith("mark ")) {
+                int markIndex = -1 + Integer.parseInt(line.substring(5));
```
maybe place the -1 after the Integer.parseInt(line.substring(5))

sean6369 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,35 @@
+    public void setDone(boolean done) {
+        isDone = done;
+    }
+}
```
more methods can be create eg. list(), add(), mark(), unmark() , for easier readability

sean6369 commented on src/main/java/Bruh.java:
```diff
@@ -0,0 +1,51 @@
+
+public class Bruh {
+    public static void main(String[] args) {
+        String logo = "Hello! I'm:\n" +
```
can set this at a constant, with all caps case variable name: LOGO

sean6369 reviewed (commented):
good job!!

gordonajajar committed:
Add increment: level-4 with TextUiTesting

Add Todo, Deadline, Event subclasses of Task. Refactor code into switch case

namsengi11 commented on src/main/java/Bruh.java:
```diff
@@ -0,0 +1,74 @@
+
+        Scanner in = new Scanner(System.in);
+        String line;
+        Task[] tasks = new Task[100];
```
100 is a magic number. Could factor it out as a constant like MAX_TASKS_LENGTH

namsengi11 commented on src/main/java/Bruh.java:
```diff
@@ -0,0 +1,74 @@
+            String[] input = line.split(" ", 2);
+            String command = input[0].toLowerCase();
+
+            if (parseInput(command, tasks, input)) return;
```
It is clever to use output of parseInput to continue/break in the main loop, but it may not be clear at first glance that parseInput method will return true only when the user inputs 'bye'

namsengi11 commented on src/main/java/Bruh.java:
```diff
@@ -0,0 +1,74 @@
+import java.sql.SQLOutput;
```
Unused import?

namsengi11 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,15 @@
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+        System.out.println("Added deadline: \n" + this.toString() + ".\n" + "You now have " + Task.getNumberOfTasks() + " tasks.");
```
You might not want to print to the terminal every time instance of Deadline is created, because you may have to instantiate objects internally without user's request.

gordonajajar committed:
Add increment: level-5

gordonajajar committed:
Merge branch 'branch-Level-5'

gordonajajar committed:
Add increment: a-packages

Also factor terminal printing during task creation out of constructors to Parser function

gordonajajar committed:
Merge branch 'branch-A-Packages'

gordonajajar committed:
Add increment: level-6 with collections ArrayList

gordonajajar committed:
Add saveTasksToDisk function

gordonajajar committed:
Add loadTasksFromDisk function

gordonajajar committed:
Merge branch 'branch-Level-6'

gordonajajar committed:
Merge branch 'branch-Level-7'

gordonajajar committed:
Refactor task loading and saving into TaskStorage class in package
storage

gordonajajar committed:
Add jar manifest

gordonajajar committed:
Fix deleting task not updating data

gordonajajar committed:
Add increment: level-8

gordonajajar committed:
Add increment: level-9 and javadoc

gordonajajar committed:
Add User Guide

gordonajajar committed:
Edited User Guide

gordonajajar committed:
chore(branch-A-JavaDoc): create empty commit so merge shows non-FF

gordonajajar committed:
Merge branch 'branch-A-JavaDoc'

gordonajajar committed:
chore(branch-Level-9): create empty commit so merge shows non-FF

gordonajajar committed:
Merge branch 'branch-Level-9'

gordonajajar committed:
Edit User Guide format

gordonajajar committed:
Commit to practice merging PR; edit user guide

gordonajajar committed:
Merge pull request #1 from gordonajajar/branch-Practice-Merge-PR

Commit to practice merging PR; edit user guide

gordonajajar committed:
Change EmptyDescriptionException print

gordonajajar committed:
Merge branch 'master' of github.com:gordonajajar/ip

gordonajajar committed:
Final iP Refactor: add Parser class

gordonajajar committed:
Remove unnecessary imports

gordonajajar committed:
Merge pull request #2 from gordonajajar/bruh-refactor-parser

Refactor: add Parser class
